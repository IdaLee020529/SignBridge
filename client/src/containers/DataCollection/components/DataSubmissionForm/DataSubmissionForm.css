.dataForm {
    width: 80%;
}

.dataForm-header-container {
    margin: 30px auto 20px auto;
    justify-items: center;
    text-align: center;
    background-color: #F6F7F9;
    padding: 50px 22px 40px 22px;
    border-radius: 8px;
}

.dataForm-header h1 {
    font-size: 48px;
    font-family: "Plus Jakarta Sans", "Plus Jakarta Sans Placeholder", sans-serif;
    font-weight: 500;
    font-style: normal;
}

.dataForm-header p {
    margin-top: 12px;
    color: rgb(115, 115, 115);
    font-family: "Plus Jakarta Sans", "Plus Jakarta Sans Placeholder", sans-serif;
    font-weight: 500;
}

.dataForm-cover {
    display: flex;
    box-shadow: -5px 3px 10px rgba(0, 0, 0, 0.4);
    max-width: 80%;
    text-align: center;
    width: 80%;
    margin: 30px auto 20px auto;
}

.dataForm-card {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-start;
    padding: 30px;
    width: 35%;
}

.dataForm-card.public {
    background-color: #dce1ec94;
}

.dataForm-card.signexpert {
    background-color: #dce1ec94;
}

.dataForm-card h1,
.dataForm-card h3 {
    color: #363636;
    margin: 15px 0
}

.dataForm-card .dataForm-card-content {
    margin-top: 20px;
}

.dataForm-card-info {
    display: flex;
    text-align: left;
    align-items: flex-start;
    margin: 25px 0;
}

.dataForm-card-info2 {
    display: flex;
    text-align: left;
    align-items: center;
    margin: 20px 0;
}


.dataForm-card-info p,
.dataForm-card-info2 p {
    font-size: 18px;
    max-width: 80%;
}

.dataForm-container {
    width: 65%;
    padding: 10px 30px;
    background-color: white;
}

.dataForm-info-block {
    display: grid;
}

.dataForm-info {
    font-size: 40px;
    justify-self: end;
    top: 0;
    right: 0;
}

.dataForm-container .input-container {
    margin-top: 10px;
    width: 100%
}

.dataForm-container h1 {
    display: block;
    margin: 0 auto;
}

.button-container {
    display: flex;
    justify-content: space-evenly;
    margin-top: 40px;
    margin-bottom: 10px;
}

.video-container {
    margin-top: 30px;
}