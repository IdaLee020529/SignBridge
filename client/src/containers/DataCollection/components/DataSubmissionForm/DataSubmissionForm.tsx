import "regenerator-runtime/runtime";
import React, { useState, useEffect } from "react";
import InputField from "../../../../components/InputField/InputField";
import { Button } from "../../../../components/Button/Button";
import VideoInput from "../../../../components/VideoInput/VideoInput";
import EmailIcon from "../EmailIcon/EmailIcon";
import PhoneIcon from "../PhoneIcon/PhoneIcon";
import InfoIcon from "../InfoIcon/InfoIcon";
import LocationIcon from "../LocationIcon/LocationIcon";
import { submitForm } from "../../../../services/dataset.service";
import "./DataSubmissionForm.css";
import Cookies from "js-cookie";
import {
    CreateNotification,
    GetUserIdByEmail,
} from "../../../../services/notification.service";
import { useTranslation } from "react-i18next";
import { toast } from "react-hot-toast";

import SpeechRecognition, {
    useSpeechRecognition,
} from "react-speech-recognition";
interface DataSubmissionFormProps {
    user: string;
}

const DataSubmissionForm: React.FC<DataSubmissionFormProps> = ({ user }) => {
    const { t, i18n } = useTranslation();
    const [isListening, setIsListening] = useState(false);

    const { transcript, resetTranscript, browserSupportsSpeechRecognition } =
        useSpeechRecognition({});

    const renderMicrophoneButton = () => {
        if (browserSupportsSpeechRecognition) {
            return (
                <button
                    className="avatar-microphone-btn"
                    onClick={() => {
                        if (!isListening) {
                            resetTranscript();
                            setCustomTranscript("");
                            SpeechRecognition.startListening({
                                language: "ms-MY",
                                continuous: true,
                            });
                            setIsListening(true);
                            toast("Listening", {
                                icon: "ðŸŽ¤",
                                style: {
                                    borderRadius: "10px",
                                    background: "#333",
                                    color: "#fff",
                                },
                            });
                        } else {
                            SpeechRecognition.stopListening();
                            setIsListening(false);
                            toast("Stopped", {
                                icon: "âœ‹",
                                style: {
                                    borderRadius: "10px",
                                    background: "#333",
                                    color: "#fff",
                                },
                            });
                        }
                    }}
                >
                    <i
                        className={`fa ${
                            isListening ? "fa-stop faStopBtn" : "fa-microphone"
                        }`}
                    ></i>
                </button>
            );
        } else {
            return (
                <button
                    className="avatar-microphone-btn disabled"
                    disabled={true}
                >
                    <i className="fa fa-microphone"></i>
                    <span className="tooltip2">
                        Voice input isn't supported on this browser
                    </span>
                </button>
            );
        }
    };

    const [customTranScript, setCustomTranscript] = useState("");

    useEffect(() => {
        setText(transcript);
    }, [transcript]);

    //Modal Control (Onsubmit popup)
    const [isModalOpen, setIsModalOpen] = useState(false);
    const handleOpenModal = () => {
        setIsModalOpen(true);
    };

    const handleCloseModal = () => {
        setIsModalOpen(false);
    };
    //Video Control
    const [videoInfo, setVideoInfo] = useState(null);
    const [resetVideo, setResetVideo] = useState(false);
    const handleVideoReset = () => {
        setResetVideo(true); // Reset resetVideo state
    };

    const handleReset = () => {
        setName("");
        setEmail("");
        setText("");
        setNameError("");
        setEmailError("");
        setTextError("");
        setResetVideo(true); // Set resetVideo to true to trigger video input reset
    };

    //Validation Control
    const validateName = (value: string) => {
        if (!value.trim()) {
            setNameError("Name is required");
            return "Name is required";
        }
        setNameError("");
        return undefined;
    };

    const validateEmail = (value: string) => {
        if (!value.trim()) {
            setEmailError("Email is required");
            return "Email is required";
        }
        // Email format validation using regex
        const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
        if (!emailRegex.test(value)) {
            setEmailError("Invalid email format");
            return "Invalid email format";
        }
        setEmailError("");
        return undefined;
    };

    const validateText = (value: string) => {
        if (!value.trim()) {
            setTextError("Text/Sentence is required");
            return "Text/Sentence is required";
        }
        setTextError("");
        return undefined;
    };

    //Form Element Setup
    const [email, setEmail] = useState("");
    const [emailError, setEmailError] = useState("");
    const handleEmailChange = (
        e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
    ) => {
        setEmail(e.target.value);
        validateEmail(e.target.value);
    };

    const [name, setName] = useState("");
    const [nameError, setNameError] = useState("");
    const handleNameChange = (
        e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
    ) => {
        setName(e.target.value);
        validateName(e.target.value);
    };

    const [text, setText] = useState("");
    const [textError, setTextError] = useState("");
    const handleTextChange = (
        e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
    ) => {
        setText(e.target.value);
        validateText(e.target.value);
    };

    //Submit Control
    const handleSubmit = async (
        e: React.FormEvent<HTMLFormElement>
    ): Promise<void> => {
        e.preventDefault(); // Prevent default form submission behavior
        if (
            nameError.length === 0 &&
            emailError.length === 0 &&
            textError.length === 0 &&
            videoInfo != null
        ) {
            const user_id = Cookies.get("user_id");
            if (!user_id) {
                console.log("Please login");
            } else {
                const status_SE = "New";
                const status_Admin = "-";
                const formData = new FormData();
                formData.append("user_id", user_id);
                formData.append("name", name);
                formData.append("email", email);
                formData.append("text_sentence", text);
                formData.append("status_SE", status_SE);
                formData.append("status_Admin", status_Admin);
                if (videoInfo) {
                    formData.append("video", videoInfo);
                    try {
                        await submitForm(formData);
                    } catch (error: any) {
                        console.error("Error");
                    }
                    handleReset();
                    await handleOpenModal();
                }

                // Send notification
                try {
                    const notificationData = {
                        receiver_id: 2,
                        sender_id: user_id ? parseInt(user_id) : 0,
                        message_en: "has submitted new text.",
                        message_bm: "telah menghantar teks baru.",
                        sign_text: text,
                        status: 0,
                        type: "New Text",
                        type_value: "newtext",
                        created_at: new Date().toISOString(),
                    };
                    console.log("notificationUserId", user_id);
                    await CreateNotification(notificationData);
                    toast.success(t("notifSuccess"));
                } catch (error) {
                    console.error("Error sending notification:", error);
                    toast.error(t("notifFailed"));
                }
            }
        }
    };

    return (
        <div className="dataForm">
            <div className="dataForm-header-container">
                <div className="dataForm-header">
                    <h1>{t("dataset_collection_form")}</h1>
                    <p>{t("datasetmsg")}</p>
                </div>
            </div>
            <div className="dataForm-cover">
                <div className={`dataForm-card ${user}`}>
                    <h1>{t("in_touch")}</h1>
                    <h3>{t("more_suggestions")}</h3>
                    <div className={"dataForm-card-content"}>
                        <div className="dataForm-card-info">
                            <LocationIcon />
                            <p> {t("neoun_address")}</p>
                        </div>
                        <div className="dataForm-card-info2">
                            <PhoneIcon />
                            <p>{t("neoun_phone")}</p>
                        </div>
                        <div className="dataForm-card-info2">
                            <EmailIcon />
                            <p>{t("neoun_email")}</p>
                        </div>
                        <div className="dataForm-card-info"></div>
                    </div>
                </div>
                <div className="dataForm-container">
                    <div className="dataForm-info-block">
                        <InfoIcon />
                    </div>
                    <div className="row justify-content-center input-container">
                        <div className="col-md-8">
                            <form onSubmit={handleSubmit} noValidate>
                                <InputField
                                    label={t("name_input")}
                                    name="name"
                                    value={name}
                                    onChange={handleNameChange}
                                    error={nameError}
                                />
                                <InputField
                                    label={t("email_input")}
                                    name="email"
                                    type="email"
                                    value={email}
                                    onChange={handleEmailChange}
                                    error={emailError}
                                />
                                <div className="voiceToTextBox">
                                    <InputField
                                        label={t("text_sentence_input")}
                                        name="text"
                                        value={text}
                                        onChange={handleTextChange}
                                        multipleLines={true}
                                        error={textError}
                                    />
                                    {renderMicrophoneButton()}{" "}
                                </div>

                                <div className="video-container">
                                    <VideoInput
                                        reset={resetVideo}
                                        onReset={handleVideoReset}
                                        setVideoInfo={setVideoInfo}
                                    />
                                </div>
                                <div className="button-container">
                                    <Button
                                        type="button"
                                        onClick={handleReset}
                                        buttonStyle="btn--reset"
                                        buttonSize="btn--large"
                                    >
                                        {t("reset_btn")}
                                    </Button>
                                    <Button
                                        type="submit"
                                        buttonStyle="btn--submit"
                                        buttonSize="btn--large"
                                    >
                                        {t("submit_btn")}
                                    </Button>
                                </div>
                            </form>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default DataSubmissionForm;
