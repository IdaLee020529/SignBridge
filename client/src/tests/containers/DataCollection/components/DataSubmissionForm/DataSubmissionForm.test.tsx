import React from "react";
import DataSubmissionForm from "../../../../../containers/DataCollection/components/DataSubmissionForm/DataSubmissionForm";
import { cleanup } from "@testing-library/react";
import { create } from "react-test-renderer";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
}));
// Mock the useSpeechRecognition hook
jest.mock("react-speech-recognition", () => ({
  useSpeechRecognition: () => ({
    transcript: "Hello, World!",
    resetTranscript: jest.fn(),
    browserSupportsSpeechRecognition: true,
  }),
}));
describe("Test DataSubmissionForm", () => {
  afterEach(cleanup); //Unmount all react tree after rendering
  const props = {
    user: "public",
    isSubmitModalOpen: false,
    showPopup: false,
    setShowPopup: jest.fn(),
    showInstructionPopup: false,
    setShowInstructionPopup: jest.fn(),
    onOpenModal: jest.fn(),
  };

  it("should render correctly", () => {
    const tree = create(<DataSubmissionForm {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
