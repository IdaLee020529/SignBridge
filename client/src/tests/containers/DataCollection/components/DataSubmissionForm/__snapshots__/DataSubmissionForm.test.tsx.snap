// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DataSubmissionForm should render correctly 1`] = `
[
  <div
    className="popup-confirmation "
  >
    <div
      className="popup-confirmation-content-header"
    >
      <h2>
        Are you sure?
      </h2>
    </div>
    <div
      className="popup-confirmation-content"
    >
      <div
        className="popup-confirmation-details"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-175ib4q-MuiPaper-root-MuiTableContainer-root"
        >
          <table
            className="MuiTable-root css-rqglhn-MuiTable-root"
            role={null}
          >
            <tbody
              className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root css-ys9ggv-MuiTableRow-root"
                role={null}
              >
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                >
                  <b>
                    Name
                  </b>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-3urw24-MuiTableCell-root"
                />
              </tr>
              <tr
                className="MuiTableRow-root css-2qi3sl-MuiTableRow-root"
                role={null}
              >
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                >
                  <b>
                    Email
                  </b>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
                />
              </tr>
              <tr
                className="MuiTableRow-root css-ys9ggv-MuiTableRow-root"
                role={null}
              >
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                >
                  <b>
                    Text
                  </b>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
                >
                  Hello, World!
                </td>
              </tr>
              <tr
                className="MuiTableRow-root css-1yq2m0q-MuiTableRow-root"
                role={null}
              >
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                >
                  <b>
                    Video
                  </b>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
                />
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        className="popup-confirmation-content-button"
      >
        <button
          className="btn btn--reset btn--large"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="ant-btn css-dev-only-do-not-override-42nv3w ant-btn-default btn-submit btn-large"
          disabled={false}
          onClick={[Function]}
          style={
            {
              "fontSize": "20px",
              "height": "auto",
              "position": "relative",
            }
          }
          type="button"
        >
          <span>
            submit_btn
          </span>
        </button>
      </div>
    </div>
  </div>,
  <div
    className="instruction-popup  "
  >
    <div
      className="instruction-popup-header"
    >
      <h1
        className="instruction-title"
      >
        Instructions
      </h1>
      <i
        className="fa fa fa-close"
        onClick={[Function]}
      />
    </div>
    <div
      className="instruction-popup-details"
    >
      <div
        className="instruction-popup-details-section1"
      >
        <p>
          <strong>
            1.
          </strong>
           Login a user account
        </p>
        <p>
          <strong>
            2.
          </strong>
           Fill up the details - Name, Email, Text/ Sentence
        </p>
        <p>
          <strong>
            3.
          </strong>
           Upload a demonstration video of the text/sentence
        </p>
        <p>
          <strong>
            4.
          </strong>
           Press the reset button to reset the form or the submit button to submit the form
        </p>
      </div>
      <hr
        className="line-separator"
      />
      <div
        className="instruction-popup-details-section2"
      >
        <p>
          Note:
        </p>
        <div
          className="instruction-popup-content"
        >
          <ul>
            <li>
              The video submitted must be in mp4 format
            </li>
            <li>
              The video should include either half or entire body of the presenter
            </li>
            <li>
              The video will only be used for avatar preparation purpose only
            </li>
          </ul>
        </div>
      </div>
      <div
        className="instruction-popup-details-section3"
      >
        <p>
          You can refer to the image below:
        </p>
        <div
          className="instruction-popup-details-image"
        >
          <img
            alt="instruction-image"
            className="instruction-image"
            src="default"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="dataForm"
  >
    <div
      className="dataForm-header-container"
    >
      <div
        className="dataForm-header"
      >
        <h1>
          dataset_collection_form
        </h1>
        <p>
          datasetmsg
        </p>
      </div>
    </div>
    <div
      className="dataForm-cover"
    >
      <div
        className="dataForm-card public"
      >
        <h1>
          in_touch
        </h1>
        <h3>
          more_suggestions
        </h3>
        <div
          className="dataForm-card-content"
        >
          <div
            className="dataForm-card-info"
          >
            <svg
              fill="currentColor"
              height="2em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": "#000",
                  "marginRight": "5px",
                }
              }
              viewBox="0 0 24 24"
              width="2em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 1.74.5 3.37 1.41 4.84.95 1.54 2.2 2.86 3.16 4.4.47.75.81 1.45 1.17 2.26.26.55.47 1.5 1.26 1.5s1-.95 1.25-1.5c.37-.81.7-1.51 1.17-2.26.96-1.53 2.21-2.85 3.16-4.4C18.5 12.37 19 10.74 19 9c0-3.87-3.13-7-7-7zm0 9.75a2.5 2.5 0 0 1 0-5 2.5 2.5 0 0 1 0 5z"
              />
            </svg>
            <p>
               
              neoun_address
            </p>
          </div>
          <div
            className="dataForm-card-info2"
          >
            <svg
              fill="currentColor"
              height="2em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": "#000",
                  "marginRight": "5px",
                }
              }
              viewBox="0 0 24 24"
              width="2em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
              />
            </svg>
            <p>
              neoun_phone
            </p>
          </div>
          <div
            className="dataForm-card-info2"
          >
            <svg
              fill="currentColor"
              height="30px"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": "#000",
                  "marginRight": "5px",
                }
              }
              viewBox="0 0 24 24"
              width="30px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
            <p>
              neoun_email
            </p>
          </div>
          <div
            className="dataForm-card-info"
          />
        </div>
      </div>
      <div
        className="dataForm-container"
      >
        <div
          className="dataForm-info-block"
        >
          <svg
            className="dataForm-info"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": "#000",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 4v16a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1V4a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1zm8 3h2v2h-2V7zm0 4h2v6h-2v-6z"
            />
          </svg>
        </div>
        <div
          className="row justify-content-center input-container"
        >
          <div
            className="col-md-8"
          >
            <div
              className="form-group "
            >
              <input
                className="form-control "
                data-testid="name"
                name="name"
                onChange={[Function]}
                placeholder=" "
                type="text"
                value=""
              />
              <label
                className="input-label "
              >
                name_input
              </label>
              <div
                className="form-error "
              />
            </div>
            <div
              className="form-group "
            >
              <input
                className="form-control "
                data-testid="email"
                name="email"
                onChange={[Function]}
                placeholder=" "
                type="email"
                value=""
              />
              <label
                className="input-label "
              >
                email_input
              </label>
              <div
                className="form-error "
              />
            </div>
            <div
              className="voiceToTextBox"
            >
              <div
                className="form-group "
              >
                <textarea
                  className="form-control  multipleLines"
                  data-testid="text"
                  name="text"
                  onChange={[Function]}
                  placeholder=" "
                  value="Hello, World!"
                />
                <label
                  className="input-label multiple_label"
                >
                  text_sentence_input
                </label>
                <div
                  className="form-error "
                />
              </div>
              <button
                className="avatar-microphone-btn"
                onClick={[Function]}
              >
                <i
                  className="fa fa-microphone"
                />
              </button>
               
            </div>
            <div
              className="video-container"
            >
              <div
                className="videoinput-class"
              >
                <span
                  className="ant-upload-wrapper css-dev-only-do-not-override-42nv3w"
                >
                  <div
                    className="ant-upload ant-upload-select"
                  >
                    <span
                      className="ant-upload"
                      onClick={[Function]}
                      onDragOver={[Function]}
                      onDrop={[Function]}
                      onKeyDown={[Function]}
                    >
                      <input
                        accept=".mp4"
                        disabled={false}
                        multiple={false}
                        onChange={[Function]}
                        onClick={[Function]}
                        style={
                          {
                            "display": "none",
                          }
                        }
                        type="file"
                      />
                      <button
                        className="ant-btn css-dev-only-do-not-override-42nv3w ant-btn-default ant-btn-lg"
                        disabled={false}
                        onClick={[Function]}
                        style={{}}
                        type="button"
                      >
                        <span
                          className="ant-btn-icon"
                          style={{}}
                        >
                          <span
                            aria-label="upload"
                            className="anticon anticon-upload"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="upload"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                              />
                            </svg>
                          </span>
                        </span>
                        <span>
                          choose_a_video
                        </span>
                      </button>
                    </span>
                  </div>
                </span>
              </div>
            </div>
            <div
              className="button-container"
            >
              <button
                className="btn btn--reset btn--large"
                onClick={[Function]}
                type="button"
              >
                reset_btn
              </button>
              <button
                className="btn btn--submit btn--large"
                onClick={[Function]}
                type="submit"
              >
                submit_btn
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
